javascript/ql/lib/semmle/javascript/frameworks/Knex.qll

```diff
@@ -46,17 +46,38 @@ module Knex {
     RawKnexSqlString() { this = any(RawKnexCall call).getArgument(0).asExpr() }
   }
 
-  /** A call that triggers a SQL query submission. */
-  private class KnexDatabaseAccess extends DatabaseAccess {
-    KnexDatabaseAccess() {
+  /** A call that triggers a SQL query submission by calling then/stream/asCallback. */
+  private class KnexDatabaseCallback extends DatabaseAccess, DataFlow::CallNode {
+    KnexDatabaseCallback() {
       this = knexObject().getMember(["then", "stream", "asCallback"]).getACall()
+    }
+
+    override DataFlow::Node getAResult() { 
+      this = knexObject().getMember("then").getACall() and
+        result = this.getCallback(0).getParameter(0)
       or
-      exists(AwaitExpr await |
-        this = await.flow() and
-        await.getOperand() = knexObject().getAUse().asExpr()
-      )
+        this = knexObject().getMember("asCallback").getACall() and
```

Comment:
```suggestion
      member = "asCallback" and
```
(Belongs with the above suggestion)

It also look like you need to run the autoformatter on this code (Ctrl + Shift + P -> `Format document`).